daml-lf/interpreter/src/main/scala/com/digitalasset/daml/lf/speedy/Compiler.scala

```diff
@@ -776,6 +783,7 @@ final case class Compiler(packages: PackageId PartialFunction Package) {
         Map.empty,
         0,
         withEnv { _ =>
+          env = env.incrPos // byKey flag
```

Comment:
Do you want to update the comment above how choices are compiled, i.e., add a byKey parameter to the abstraction?

Reply:
I was a bit too quick to add you as reviewer. This was done in the previous PR. It is rebase now.

Reply:
On the other hand, why do you even need to pass the `byKey` flag through the compiled DAML code? Whether this is a byKey choice is known statically. So if you move the `byKey` flag into the first argument list for `SBUBeginExercise`, you shouldn't have to pass this around. But you know the internals better than I do.

Reply:
A choice is  compiled into a speedy function when the archive is loaded (see the function `compileChoice`). The resulting function is used for both kind of exercises (see `compileExercise` and `compileExerciseByKey`). The changes you advocate needs deeper refactoring; this probably needs to generate two functions for each choice a template with key.
I am not sure it is worthwhile.

Reply:
Thanks for the explanation. And yes, you'd need to functions for implementing my proposal; I didn't know about the load-vs-run distinction. My proposal then boils down to specializing the current function into two. We have other performance bottlenecks to sort out first.